xapian-applications/omega/query.cc

```diff
@@ -1558,7 +1558,6 @@ eval(const string &fmt, const vector<string> &param)
 		string data = args[0];
 		string hash = "md5";
 		if (args.size() > 1) hash = args[1];
-		std::transform(hash.begin(), hash.end(), hash.begin(), ::tolower);
```

Comment:
> While only recognising md5 in lower case seems reasonable to me (in general, OmegaScript is case sensitive, and we expect lower case tokens for things like query parser flags, prefix settings, etc), the documentation ought to be clear on this.

Should I include this `std::transform` operation back here with a note in the documentation that `HASH` must be lowercase?

> Also, quietly producing empty output if the algorithm name isn't recognised doesn't seem helpful, especially as it'll mean everything will appear to hash the same. I think that should be an error.

Okay, I'll add a `Xapian::InvalidOperationError` to raise an error if the algorithm name is invalid or not recognised.